tests/test_file_cache_path.py

```diff
@@ -0,0 +1,646 @@
+    assert str(FCPath('/a/b') / '/c') == '/c'
+    assert str(FCPath('/a/b') / 'c' / 'http://bucket/x' / 'y') == 'http://bucket/x/y'
+    assert str(FCPath('/a') / Path('b', 'c') / FCPath('d/e')) == '/a/b/c/d/e'
+    with FileCache('test', delete_on_exit=True) as fc:
+        p = FCPath('a', filecache=fc, anonymous=True, lock_timeout=59, nthreads=2)
+        p2 = p / 'c'
+        assert str(p2) == 'a/c'
+        assert p2._filecache is fc
+        assert p2._anonymous
+        assert p2._lock_timeout == 59
+        assert p2._nthreads == 2
+
+
+def test_rtruediv():
+    assert str('a' / FCPath('b/c')) == 'a/b/c'
+    assert str('/a' / FCPath('b/c')) == '/a/b/c'
+    assert str('/a' / FCPath('b')) == '/a/b'
+    with FileCache('test', delete_on_exit=True) as fc:
+        p = FCPath('c', filecache=fc, anonymous=True, lock_timeout=59, nthreads=2)
+        p2 = 'a' / p
+        assert str(p2) == 'a/c'
+        assert p2._filecache is fc
+        assert p2._anonymous
+        assert p2._lock_timeout == 59
+        assert p2._nthreads == 2
+        p3 = FCPath('a', lock_timeout=100, nthreads=9) / p
+        assert str(p3) == 'a/c'
+        assert p3._filecache is None
+        assert not p3._anonymous
+        assert p3._lock_timeout == 100
+        assert p3._nthreads == 9
+        assert str('http://bucket/a' / FCPath('b/c') / 'd' / FCPath('e')) == \
+            'http://bucket/a/b/c/d/e'
+
+
+def test_parent():
+    p = FCPath('http://bucket/a/b/c', nthreads=3)
+    p2 = p.parent
+    assert isinstance(p2, FCPath)
+    assert str(p2) == 'http://bucket/a/b'
+    assert p2._nthreads == 3
+    p3 = p2.parent
+    assert isinstance(p3, FCPath)
+    assert str(p3) == 'http://bucket/a'
+    assert p3._nthreads == 3
+    p4 = p3.parent
+    assert isinstance(p4, FCPath)
+    assert str(p4) == 'http://bucket'
+    assert p4._nthreads == 3
+    p5 = p4.parent
+    assert isinstance(p5, FCPath)
+    assert str(p5) == 'http://bucket'
+    assert p5._nthreads == 3
+
+
+def test_parents():
+    p = FCPath('http://bucket/a/b/c', nthreads=3)
+    p2 = p.parents
+    assert all([isinstance(x, FCPath) for x in p2])
+    assert all([x._nthreads == 3 for x in p2])
+    assert [str(x) for x in p2] == ['http://bucket/a/b', 'http://bucket/a',
+                                    'http://bucket']
+
+
+def test_match():
+    p = FCPath('abc/def')
+    with pytest.raises(ValueError):
+        p.match('')
+    assert p.match('def')
+    assert not p.match('deF')
+    assert p.match(FCPath('def'))
+    assert not p.match('f')
+    assert p.match('*f')
+    assert not p.match('c/def')
+    assert not p.match('/*f')
+    assert p.match('*/*f')
+    assert not p.match('/def')
+    assert p.match('/*/def')
+    assert p.match('/abc/def')
+    assert not p.match('/ABC/def')
+    assert not p.match('/zz/abc/def')
+    p = FCPath('C:/a/b')
+    assert p.match('b')
+    assert p.match('a/b')
+    assert p.match('c:/a/b')
+    assert p.match('C:/a/b')
+    assert p.match('C:/a/b')
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider removing duplicate test assertion</b></div>
<div id="fix">

Consider removing one of the duplicate assertions `assert p.match('C:/a/b')` as it appears twice consecutively and tests the same condition. A similar issue was also found in tests/test_file_cache_path.py (line 494-495).
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -487,3 +487,2 @@
      assert p.match('C:/a/b')
 -    assert p.match('C:/a/b')
      assert not p.match('d:/a/b')
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/24#issuecomment-2495649557>#77344d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
Good suggestion.